### PR TITLE
[util] Add C Enum Support to sparse-fsm-encode.py

### PR DESCRIPTION
**Pull request by lenary**
This adds support for generating C enums to the sparse encoder script.
We intend to use this when hardening the software we're writing in C,
including in particular the Mask ROM.

The script has not been improved much, as the intention is this is run
once for a specific enum, and the result is copied into a C file and
modified, rather than the script producing a perfect enum when run.

### CODE REVIEW

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (gkelly) [inline hw/ip/prim/util/sparse-fsm-encode.py:63]: Seems a bit cruel to take the HW tool and change to output C by default. :P

**Review by gkelly — APPROVED** [review]
Needs review from HW before submitting.

**Review by msfschaffner — COMMENTED** [review]
Hey @lenary, this LGTM. 

Note however, that there is another PR in flight that expands the error, help and template messages: https://github.com/lowRISC/opentitan/pull/3760. It would be good if you could base this PR on that change...

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (msfschaffner) [reply hw/ip/prim/util/sparse-fsm-encode.py:63]: Can we make this SV by default? I think we're going to use it more widely for HW ;)...

Comment (lenary) [reply hw/ip/prim/util/sparse-fsm-encode.py:63]: Yeah this is an oops. The intention is that it's sv by default (and you'll see i didn't add the `--language` to the sv template).

**Comment by lenary** [comment]
@msfschaffner this now includes the changes from #3760, and so is blocked on that PR. I'll rebase if any other changes happen, in the meantime, I will mark this as a draft.

**Comment by msfschaffner** [comment]
Great, thanks!

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (msfschaffner) [inline hw/ip/prim/util/sparse-fsm-encode.py:244]: not sure, but you may also want to print out some additional guidance here to instruct the user to copy the whole comment for reproducibility / auditability (as in the RTL case)...

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (msfschaffner) [inline hw/ip/prim/util/sparse-fsm-encode.py:245]: just noting that this line can get really long, depending on the seed (and that could cause lint issues).
so we may have to split this onto two lines

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (lenary) [reply hw/ip/prim/util/sparse-fsm-encode.py:244]: I've made something like your instructions and we print it out now.

File: hw/ip/prim/util/sparse-fsm-encode.py
Comment (msfschaffner) [inline hw/ip/prim/util/sparse-fsm-encode.py:31]: nit: can you update the usage string as well?

Comment (lenary) [reply hw/ip/prim/util/sparse-fsm-encode.py]: Done, using argparse's support for this rather than a hard-coded string.

**Pull request merged**
